src/Sylius/Component/Core/spec/Taxation/OrderShipmentTaxesByZoneApplicatorSpec.php

```diff
@@ -81,6 +82,34 @@ function it_applies_shipment_taxes_on_order_based_on_shipment_adjustments_and_ra
         $this->apply($order, $zone);
     }
 
+    function it_does_nothing_if_there_are_no_shipment_taxes_on_order(
+        $adjustmentsFactory,
+        $calculator,
+        $taxRateResolver,
+        AdjustmentInterface $shippingAdjustment,
+        Collection $shippingAdjustments,
+        OrderInterface $order,
+        ShipmentInterface $shipment,
+        ShippingMethodInterface $shippingMethod,
+        TaxRateInterface $taxRate,
+        ZoneInterface $zone
+    ) {
+        $order->getLastShipment()->willReturn($shipment);
+        $shipment->getMethod()->willReturn($shippingMethod);
+        $taxRateResolver->resolve($shippingMethod, ['zone' => $zone])->willReturn($taxRate);
+
+        $order->getAdjustments(AdjustmentInterface::SHIPPING_ADJUSTMENT)->willReturn($shippingAdjustments);
+        $shippingAdjustments->isEmpty()->willReturn(false);
+        $shippingAdjustments->last()->willReturn($shippingAdjustment);
+        $shippingAdjustment->getAmount()->willReturn(1000);
+
+        $calculator->calculate(1000, $taxRate)->willReturn(0);
+
+        $adjustmentsFactory->createWithData(AdjustmentInterface::TAX_ADJUSTMENT, 'Simple tax (0%)', 0, false)->shouldNotBeCalled();
```

Comment:
`$adjustmentsFactory->createWithData(Argument:any())` will match `$adjustmentsFactory->createWithData('foo')` but not `$adjustmentsFactory->createWithData('foo', 'bar')`. It's better to use `$adjustmentsFactory->createWithData(Argument::cetera())` instead as it will match every `createWithData` method call.